src/garage/envs/gym_env.py

```diff
@@ -205,13 +209,21 @@ def step(self, action):
         Raises:
             RuntimeError: if `step()` is called after the environment has been
                 constructed and `reset()` has not been called.
+            RuntimeError: if underlying environment outputs inconsistent
+                env_info keys.
 
         """
         if self._step_cnt is None:
             raise RuntimeError('reset() must be called before step()!')
 
         observation, reward, done, info = self._env.step(action)
 
+        # check that env_infos are consistent
```

Comment:
Unfortunately, HalfCheetah is broken, so this change actually needs to go *after* the `TimeLimit.truncated` logic below. That should let you merge this change.